### PR TITLE
Chat

### PR DESCRIPTION
**Pull request by roymoran**

### CODE REVIEW

File: src/components/chat/newMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import { RouteComponentProps, withRouter } from 'react-router';
+import { NEWMESSAGE_PAGE_TITLE } from 'src/Const';
```
Comment (rmjordas) [inline src/components/chat/newMessage.tsx]: [0] This import fails the CI

File: src/components/user/userEdit.tsx [hunk 2]
```diff
@@ -4,6 +4,7 @@ import InputLabel from '@material-ui/core/InputLabel';
 import * as React from 'react';
 import styled from 'styled-components';
 
+import { EDITPROFILE_PAGE_TITLE } from 'src/Const';
```
Comment (rmjordas) [inline src/components/user/userEdit.tsx]: [0] Fails tests on CI

File: src/components/mentor/mentorEdit.tsx [hunk 3]
```diff
@@ -2,8 +2,8 @@ import FormControl from '@material-ui/core/FormControl';
 import Input from '@material-ui/core/Input';
 import InputLabel from '@material-ui/core/InputLabel';
 import * as React from 'react';
+import { EDITPROFILE_PAGE_TITLE } from 'src/Const';
```
Comment (rmjordas) [inline src/components/mentor/mentorEdit.tsx]: [0] Does not fail tests on my machine but should be consistent with other imports

File: src/components/mentor/mentorFeatured.tsx [hunk 4]
```diff
@@ -1,31 +1,37 @@
 import * as React from 'react';
-import { RouteComponentProps, withRouter } from 'react-router-dom';
 import styled from 'styled-components';
 
+import { FEATURED_PAGE_TITLE } from 'src/Const';
```
Comment (rmjordas) [inline src/components/mentor/mentorFeatured.tsx]: [0] Does not fail tests on my machine but should be consistent with other imports

File: src/components/mentor/mentorSettings.tsx [hunk 5]
```diff
@@ -3,12 +3,14 @@ import ListItem from '@material-ui/core/ListItem';
 import ListItemText from '@material-ui/core/ListItemText';
 import SvgIcon from '@material-ui/core/SvgIcon';
 import * as React from 'react';
-import { RouteComponentProps, withRouter } from 'react-router';
 import { Link } from 'react-router-dom';
+import { SETTINGS_PAGE_TITLE } from 'src/Const';
```
Comment (rmjordas) [inline src/components/mentor/mentorSettings.tsx]: [0] Fails tests on CI

File: src/components/user/userFeatured.tsx [hunk 6]
```diff
@@ -1,31 +1,41 @@
 import * as React from 'react';
-import { RouteComponentProps, withRouter } from 'react-router-dom';
 import styled from 'styled-components';
 
+import { FEATURED_PAGE_TITLE, SEARCH_VALUE } from 'src/Const';
```
Comment (rmjordas) [inline src/components/user/userFeatured.tsx]: [0] Fails tests on CI

File: src/components/user/userSettings.tsx [hunk 7]
```diff
@@ -5,14 +5,16 @@ import ListItemText from '@material-ui/core/ListItemText';
 import SvgIcon from '@material-ui/core/SvgIcon';
 import * as nodeCrypto from 'crypto';
 import * as React from 'react';
-import { RouteComponentProps, withRouter } from 'react-router';
 import { Link } from 'react-router-dom';
 import styled from 'styled-components';
 
+import { SETTINGS_PAGE_TITLE } from 'src/Const';
```
Comment (rmjordas) [inline src/components/user/userSettings.tsx]: [0] Fails tests on CI

File: src/Const.ts [hunk 8]
```diff
@@ -34,3 +37,16 @@ export const DASHBOARD_VALUE = 'dashboard';
 export const MENTORSHIPS_VALUE = 'mentorships';
 export const MESSAGES_VALUE = 'messages';
 export const SETTINGS_VALUE = 'settings';
+export const SEARCH_VALUE = 'search';
+export const EDIT_VALUE = 'edit';
+// Styling for message status Dot on Messages page
+// and Active/Pending mentorships
+export const NEW_MESSAGE_FILL = '#60C3FF';
+export const NO_NEW_MESSAGE_FILL = '#E2E8ED';
```
Comment (rmjordas) [inline src/Const.ts]: Possible to put these constants in an object to be passed to a `styled-component`'s `<ThemeProvider>`?

```jsx
const theme = {
  newMessageFill: '#60c3ff',
  noNewMessageFill: '#32383d',
};

/* ... */

<ThemeProvider theme={theme}>
  <TheApp />
</ThemeProvider>
```

Then to access those constants:

```jsx
const StyledMessageStatus = styled.div`
   padding-right: 32px;
`;

const StyledMessageStatusIcon = styled.svg`
  fill: ${props => props.newMessage ? props.theme.newMessageFill : props.theme.noNewMessageFill}
`;

/* a very basic example */

<ThemeProvider theme={theme}>
  <StyledMessageStatus>
    <StyledMessageStatusIcon />
  </StyledMessageStatus>
</ThemeProvider>
```

This isn't necessary for this PR, btw. Just trying to get some thoughts.

Styled Components docs on theming: https://www.styled-components.com/docs/advanced#theming

File: src/components/auth/mentorLogin.tsx [hunk 9]
```diff
@@ -22,10 +18,17 @@ class MentorLogin extends React.Component<RouteComponentProps<RouteProps>> {
...
+      Code: params.get('code'),
+      SendBirdAccessToken: 'sample-access-token',
+      State: params.get('state')
+    };
+
+    auth.mentorLogin(linkedInAuthCode).then((mentorJwt: JwtAuthToken) => {
       SessionService.StoreJwt(mentorJwt);
       const mentor: IMentor = JwtTokenHelper.decodeMentorJwt(mentorJwt.Token);
       SessionService.StoreMentor(mentor);
+      SendBirdClient.connect(mentor.Id!);
```
Comment (rmjordas) [inline src/components/auth/mentorLogin.tsx]: Please correct me, but `Id` on `IMentor` is not optional or allows a `null` value so null check is probably not needed?
```suggestion
      SendBirdClient.connect(mentor.Id);
```

File: src/components/auth/mentorSignup.tsx [hunk 10]
```diff
@@ -22,14 +18,21 @@ class MentorSignup extends React.Component<RouteComponentProps<RouteProps>> {
...
+      SendBirdAccessToken: 'sample-access-token',
+      State: params.get('state')
+    };
+
+    auth.mentorSignup(linkedInAuthCode).then((mentorJwt: JwtAuthToken) => {
       SessionService.StoreJwt(mentorJwt);
       const mentor: IMentor = JwtTokenHelper.decodeMentorJwt(mentorJwt.Token);
       SessionService.StoreMentor(mentor);
-
+      SendBirdClient.connect(mentor.Id!);
```
Comment (rmjordas) [inline src/components/auth/mentorSignup.tsx]: [1] Null check question

File: src/components/chat/newMessage.tsx (hunk 1 above)
Comment (roymoran) [reply src/components/chat/newMessage.tsx]: fixed import path

File: src/components/user/userEdit.tsx (hunk 2 above)
Comment (roymoran) [reply src/components/user/userEdit.tsx]: fixed

File: src/components/auth/mentorLogin.tsx (hunk 9 above)
Comment (roymoran) [reply src/components/auth/mentorLogin.tsx]: you're right, it's not needed here. Thanks!

File: src/components/auth/mentorSignup.tsx (hunk 10 above)
Comment (roymoran) [reply src/components/auth/mentorSignup.tsx]: removed.

File: src/Const.ts (hunk 8 above)
Comment (roymoran) [reply src/Const.ts]: Agreed. We can open an issue for this for a future PR.

**Pull request merged**